src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1566,6 +1569,9 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
                     span = span.Slice(current.Length);
                     if (NetEventSource.Log.IsEnabled()) s.thisRef.Trace(s.http2Stream.StreamId, $"Wrote HEADERS frame. Length={current.Length}, flags={flags}");
 
+                    // RTT PING bookeeping
+                    s.thisRef._rttEstimator.OnDataOrHeadersOrWindowUpdateSent();
```

Comment:
`OnDataOrHeadersOrWindowUpdateSent` is implemented with an interlocked operation. I assume the perf impact is marginal compared to the buffer copying code above, but I may be wrong.